### PR TITLE
Adds Intercluster Distance Map visualizer

### LINKED ISSUE #571: Intercluster Distance Map (MDS)
**Describe the solution you'd like**
Create a cluster visualization that displays the distance between cluster centers and cluster relative sizes by using [multidimensional scaling (MDS)](http://scikit-learn.org/stable/modules/generated/sklearn.manifold.MDS.html). The visualization would be:

1. A scatter plot whose points would be described by the MDS embedding of the cluster centers 
2. Each point would be sized according to the number of instances belonging tot he cluster. 
3. Each point would be labeled with its cluster index 

**Is your feature request related to a problem? Please describe.**
Between this and #570, Yellowbrick could provide a reasonable approximation of PyLDAViz in an interactive notebook with ipywidgets. 

**Examples**

screenshot 2018-08-21 06 55 38

### PR DESCRIPTION
**Pull request by bbengfort**
Intercluster distance maps display an embedding of the cluster centers in 2 dimensions with the distance to other centers preserved. E.g. the closer to centers are in the visualization, the closer they are in the original feature space. The clusters are sized according to a scoring metric. By default, they are sized by population, e.g. the number of instances that belong to each center. This gives a sense of the relative importance of clusters. Note, however, that because two clusters overlap in the 2D space, it does not imply that they overlap in the original feature space.

This PR also implements a `prop_to_size` method that can scale a set of values into a region that is suitable for displaying marker size or edge width. 

We're currently heading toward:

icdm

And we're close, however there are a few things to wrap up:

- [x] finish up the tests for MDS and TSNE embeddings 
- [x] fix the bug with the legend using ``make_axes_locatable``
- [x] cleanup the codebase (it's a bit messy)
- [x] docstring for the visualizer 
- [x] begin the documentation for the visualizer 
- [x] create an issue for extensions, e.g. setting color/opacity of clusters, other scoring methods. 
- [x] quick method 

Fixes #571

### CODE REVIEW

**Comment by bbengfort** [comment]
Oh good, according to Travis, `prop_to_size` gives two different values for Python2 and Python3 ... grr...

**Comment by bbengfort** [comment]
I've figured out how to use [inset_axes](https://matplotlib.org/2.0.2/mpl_toolkits/axes_grid/users/overview.html#insetlocator) to create custom legends so now we're on to:

icdm-inset

File: yellowbrick/cluster/icdm.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+                self.scores_[idx], (-0.30, 1-(2*scaled[idx])), xytext=(1, 1-(2*scaled[idx])),
+                arrowprops=dict(arrowstyle="wedge", color="#2e7193"), va='center', ha='center',
+            )
+
+        # Draw size legend title
+        self.lax.text(s="membership", x=0, y=1.2, va='center', ha='center')
+
+        # Ensure the current axes is always the main axes after modifying the
+        # inset axes and while drawing.
+        plt.sca(self.ax)
```
Comment (rebeccabilbro) [inline yellowbrick/cluster/icdm.py]: This membership legend is super slick!!

File: docs/api/cluster/icdm.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.. -*- mode: rst -*-
+
+Intercluster Distance Maps
+==========================
+
+Intercluster distance maps display an embedding of the cluster centers in 2 dimensions with the distance to other centers preserved. E.g. the closer to centers are in the visualization, the closer they are in the original feature space. The clusters are sized according to a scoring metric. By default, they are sized by membership, e.g. the number of instances that belong to each center. This gives a sense of the relative importance of clusters. Note however, that because two clusters overlap in the 2D space, it does not imply that they overlap in the original feature space.
+
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply docs/api/cluster/icdm.rst]: Yeh, that is a good resource - I read it while implementing this and we should reference it in the docs. I would very much like to extend the documentation with real-life examples, I think the `make_blobs` example is a good first example just to show how it works, but I'm hoping we can add an LDA example with text and the walking path example soon. 

I'd also like to include in the documentation a detailed description of how the legend works since it might not be obvious right off the bat. The three circles represent the 25th, 50th, and 75th percentile sizes of the circles drawn on the graph (scaled to the inset size).  I think having documentation about how to interpret the legend (e.g. what it means that the median is close to the 75th percentile, etc.) would also enhance the documentation.

File: yellowbrick/cluster/icdm.py [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    to centers are in the visualization, the closer they are in the original
+    feature space. The clusters are sized according to a scoring metric. By
+    default, they are sized by membership, e.g. the number of instances that
+    belong to each center. This gives a sense of the relative importance of
+    clusters. Note however, that because two clusters overlap in the 2D space,
+    it does not imply that they overlap in the original feature space.
+
+    Parameters
+    ----------
+    model : a Scikit-Learn clusterer
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/cluster/icdm.py]: Can we add this to #582 to discuss? It is a tiny bit trickier to do this than it seems on the face of it.

File: yellowbrick/cluster/icdm.py (hunk 1 above)
Comment (bbengfort) [reply yellowbrick/cluster/icdm.py]: Thanks! Probably the part I worked the longest on ... figuring out the area vs. radius thing stumped me for a long time.

File: yellowbrick/cluster/icdm.py [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+    ----------
+    cluster_centers_ : array of shape (n_clusters, n_features)
+        The computed cluster centers from the underlying model.
+
+    embedded_centers_ : array of shape (n_clusters, 2)
+        The positions of all the cluster centers on the graph.
+
+    scores_ : array of shape (n_clusters,)
+        The scores of each cluster that determine its size on the graph.
+
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/cluster/icdm.py]: Done!
